### ISSUE TITLE
IDE0004 false positive when casting int litral to nint and trying to assign to IntPtr

### ISSUE THREAD

**Issue by Tan90909090** [dotnet/roslyn#64825]
**Version Used**: 
Microsoft Visual Studio Community 2022 (64-bit) - Current
Version 17.3.6

**Steps to Reproduce**:
1. Set `<LangVersion>10.0</LangVersion>` in csproj.
2. Write a following code:
```cs
System.IntPtr x = (nint)1;
``` 

**Expected Behavior**:
There is no IDE0004.

**Actual Behavior**:
There is an `IDE0004 Cast is redundant` on `(nint)`.
If I remove the cast then I get an error: `CS0266 Cannot implicitly convert type 'int' to 'System.IntPtr'. An explicit conversion exists (are you missing a cast?)`

  Note: The cast may be really redundant when using C# 11 because there is a new feature "numeric IntPtr".

**Comment by CyrusNajmabadi** [comment]
This was changed.  This is now always legal (even on langversion10).  So it's by design that we offer to remove even then.
